sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -134,4 +143,108 @@ export class StudentController extends BaseController {
   ): Promise<void> {
     await this.studentService.synchronizeFromUserInfo(userToken);
   }
+
+  @Get("/apply-PD-status")
+  async applyForPDStatus(@UserToken() userToken: IUserToken): Promise<void> {
+    // Get student details
+    const existingStudent = await this.studentService.getStudentByUserName(
+      userToken.userName,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with the student name ${userToken.userName}`,
+      );
+    }
+    //check pd status in db, student should only allowed to check PD status once
+    // existingStudent?.StudentPDSentAt is set when student apply for PD Status first
+    // StudentPDStatus is null  before PD checker update status
+    // StudentPDStatus is true if PD Confirmed by ATBC
+    // StudentPDStatus is false if PD Denied by ATBC
+    // if student has a valid only, he/she should allow for a PD check
+    // if studentPDVerified is true from `SFASDB`, then no need to apply for PD to ATBC
+
+    if (
+      (existingStudent?.validSIN ?? false) &&
+      !existingStudent?.StudentPDSentAt &&
+      existingStudent?.StudentPDStatus === null &&
+      !existingStudent?.studentPDVerified
+    ) {
+      // create client payload
+      const payload: ATBCCreateClientPayload = {
+        SIN: existingStudent?.sin,
+        firstName: userToken?.givenNames,
+        lastName: userToken?.lastName,
+        birthDate: userToken?.birthdate,
+        email: userToken?.email,
+      };
+      // api to create student profile in ATBC
+      const response = await this.atbcService.ATBCcreateClient(payload);
+      if (response?.code === 200) {
+        //code to update PD Sent Date
+        await this.studentService.updatePDSentDate(existingStudent?.id);
+      }
+    } else {
+      let msg = "";
+      if (!(existingStudent?.validSIN ?? false))
+        msg = `${userToken.userName}'s sin is not valid`;
+      if (existingStudent?.StudentPDSentAt)
+        msg = `${userToken.userName}'s already applied for PD status`;
+      if (
+        existingStudent?.StudentPDStatus !== null ||
+        existingStudent?.studentPDVerified
+      )
+        msg = `${userToken.userName}'s PD status already captured`;
+      throw new NotAcceptableException(msg);
```

Comment:
I don`t believe that NotAcceptableException applies here, please check if it makes sense; https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406
I would recommend a UnprocessableEntityException or BadRequestException.

Reply:
In general, do we need to provide this level of detail in the exception? I would recommend log this on API and just throw the exception with a generic message.
The user is only supposed to make this call if he fits into the proper conditions to do it. The UI will not show the option if the user is not allowed to execute it.
If this method is executed and the user does not qualifies for it, it must be because he tries to bypass the UI rules in some way.